src/service/video-manager-impl.js

```diff
@@ -583,54 +666,41 @@ class VideoEntry {
       || !this.initialRect_
       || !this.internalElement_
       || (this.getPlayingState() != PlayingStates.PLAYING_MANUAL
-          && !this.internalElement_.classList.contains(DOCK_CLASS))
+              && !this.internalElement_.classList.contains(DOCK_CLASS))
     ) {
       return;
     }
 
-    // Initialize docking width/height
-    if (this.minimizePosition_ != MinimizePositions.INVIEW) {
-      this.vsync_.mutate(() => {
-        this.startDocking_();
-      });
-    }
-
-    // Temporary fix until PositionObserver somehow tracks objects outside of
-    // the viewport (forces the style to be what we want in the final state)
-    if (this.inViewportHeight_ == 0
-       && this.getPlayingState() == PlayingStates.PLAYING_MANUAL
-       && this.minimizePosition_ != MinimizePositions.DEFAULT) {
-      this.vsync_.mutate(() => {
-        this.endDocking_();
-      });
-      return;
-    }
-
-    // During the docking transition we either perform the docking or undocking
-    // scroll-bound animations
-    //
-    // Conditions for animating the video are:
-    // 1. The video is out of view and it has been in-view at least once before
-    const outOfView = this.minimizePosition_ != MinimizePositions.INVIEW
-                      && this.minimizePosition_ != MinimizePositions.DEFAULT;
-    // 2. Is either manually playing or paused while docked (so that it is
-    // undocked even when paused)
-    const manPlaying = (this.getPlayingState() == PlayingStates.PLAYING_MANUAL);
-    const paused = this.getPlayingState() == PlayingStates.PAUSED;
-    const docked = this.internalElement_.classList.contains(DOCK_CLASS);
-
-    if (outOfView && (manPlaying || (paused && docked))) {
-      // We animate docking or undocking
-      this.vsync_.mutate(() => {
-        this.animateDocking_();
-      });
-    } else if (this.minimizePosition_ == MinimizePositions.INVIEW) {
-      // Here undocking animations are done so we restore the element
-      // inline by clearing all styles and removing the position:fixed
-      this.vsync_.mutate(() => {
-        this.unDockVideo_();
-      });
-    }
+    this.vsync_.mutate(() => {
```

Comment:
split into measure and mutate. from what I see `updateDockableVideoPosition_` only does measurements so `vsync.run(measure: updateDockableVideoPosition_, mutate: // short-circut and rest of this code)`